plugins/search/src/components/SearchFilterNext/SearchFilterNext.tsx

```diff
@@ -0,0 +1,177 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, ChangeEvent, useEffect } from 'react';
+import {
+  makeStyles,
+  FormControl,
+  FormControlLabel,
+  InputLabel,
+  Checkbox,
+  Select,
+  MenuItem,
+  FormLabel,
+} from '@material-ui/core';
+
+import { useSearch } from '../SearchContext';
+
+const useStyles = makeStyles({
+  label: {
+    textTransform: 'capitalize',
+  },
+});
+
+export type Component = {
+  className?: string;
+  name: string;
+  values?: string[];
+  defaultValue?: string[] | string | null;
+};
+
+export type Props = Component & {
+  component: (props: Component) => ReactElement;
+  debug?: boolean;
+};
+
+const CheckboxFilter = ({
+  className,
+  name,
+  defaultValue,
+  values = [],
+}: Component) => {
+  const classes = useStyles();
+  const { filters, setFilters } = useSearch();
+
+  useEffect(() => {
+    if (Array.isArray(defaultValue)) {
+      setFilters(prevFilters => ({
+        ...prevFilters,
+        [name]: defaultValue,
+      }));
+    }
+  }, [name, defaultValue, setFilters]);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const {
+      target: { value, checked },
+    } = e;
+
+    setFilters(prevFilters => {
+      const { [name]: filter, ...others } = prevFilters;
+      const rest = ((filter as string[]) || []).filter(i => i !== value);
+      const items = checked ? [...rest, value] : rest;
+      return items.length ? { ...others, [name]: items } : others;
+    });
+  };
+
+  return (
+    <FormControl
+      className={className}
+      fullWidth
+      data-testid="search-checkboxfilter-next"
+    >
+      <FormLabel className={classes.label}>{name}</FormLabel>
+      {values.map((value: string) => (
+        <FormControlLabel
+          key={value}
+          control={
+            <Checkbox
+              color="primary"
+              tabIndex={-1}
+              inputProps={{ 'aria-labelledby': value }}
+              value={value}
+              name={value}
+              onChange={handleChange}
+              checked={((filters[name] as string[]) ?? []).includes(value)}
+            />
+          }
+          label={value}
+        />
+      ))}
+    </FormControl>
+  );
+};
+
+const SelectFilter = ({
+  className,
+  name,
+  defaultValue,
+  values = [],
+}: Component) => {
+  const classes = useStyles();
+  const { filters, setFilters } = useSearch();
+
+  useEffect(() => {
+    if (typeof defaultValue === 'string') {
+      setFilters(prevFilters => ({
+        ...prevFilters,
+        [name]: defaultValue,
+      }));
+    }
+  }, [name, defaultValue, setFilters]);
```

Comment:
Usually you don't want to rerender based on the default value changing ... that's a bit of a pitfall, you'd want to for example instead do

```ts
useSearch({
  [name]: defaultValue
})
```

and make sure that inside that hook, it only gives that struct as `useState(initialSearchValues)` or something, which means it'll _only_ care about what its value was on first mount.